executor/src/witgen/util.rs

```diff
@@ -1,23 +1,11 @@
-use powdr_ast::analyzed::{AlgebraicExpression as Expression, AlgebraicReference};
+use powdr_ast::analyzed::{AlgebraicExpression, AlgebraicReference};
 
-/// Checks if an expression is
-/// - a polynomial
-/// - not part of a polynomial array
-/// - not shifted with `'`
-///
-/// and return the polynomial if so
-pub fn try_to_simple_poly<T>(expr: &Expression<T>) -> Option<&AlgebraicReference> {
-    if let Expression::Reference(p @ AlgebraicReference { next: false, .. }) = expr {
+/// Checks if an algebraic expression just a polynomial / column reference without "next"
+/// and returns the polynomial if so
+pub fn try_to_simple_poly<T>(expr: &AlgebraicExpression<T>) -> Option<&AlgebraicReference> {
+    if let AlgebraicExpression::Reference(p @ AlgebraicReference { next: false, .. }) = expr {
         Some(p)
     } else {
         None
     }
 }
-
-pub fn try_to_simple_poly_ref<T>(expr: &Expression<T>) -> Option<&AlgebraicReference> {
```

Comment:
This function does exactly the same thing as the only other function in this file...